### DISCUSSION TITLE
Will Lit support Redux in the future?

### DISCUSSION THREAD

**Discussion by ouweiya** [lit/lit#4549] [General]
Lit has successfully supported `Preact Signals`, which I've tried, but it doesn't address the inadequacies of Lit in state management. Redux is very mature in state management and has a thriving ecosystem. I believe Lit should introduce its own official binding package, such as `Lit-Redux`.
I've seen some examples of interacting with `Redux` through `Reactive Controllers`, but there's a lack of a glue layer between UI components and data state, just like `React-Redux`.

**Comment by pascalvos** [comment]
It would work out of the box if you google `redux lit` there are multiple examples.
if it doesn't work you can post a playground https://lit.dev/playground/ to show what problem you run into

**Comment by augustjk** [comment]
We considered making a labs package for a reactive controller that integrates with redux. Personally I am in favor of doing that. It would likely be just a more polished version of the reactive controller examples you've seen.

I'm not sure what you mean by the "lack of a glue layer between UI components and data state" as reactive controllers will introduce the state into the component and that can be used for the UI templates.

Reactive controllers are essentially like the hooks provided by `react-redux`. I suppose it's possible to provide a mixin that would be like the higher order component model from React.

**Reply by mkutny** [reply]
You have a nice page on [Reactive Controllers](https://lit.dev/docs/composition/controllers/).

Maybe adding a Use Case on that page would be helpful.

Examples on the internet are several years old and I've got an overall feeling that they are (a bit) outdated.

**Reply by augustjk** [reply]
I'm hoping we can get this published! https://github.com/lit/lit/pull/4586

**Reply by mkutny** [reply]
It would be really lovely.

**Reply by mkutny** [reply]
Just in case, in your branch you didn't add this package to the main README of lit/lit.
